Description: Use right compiler for compiling
	Compiler use for cross compiling the source need to
	consistent with	target architecture and path to pkgconfig file
	need to consistent with host-specific

diff --git a/beav-1.40/debian/rules b/beav-1.40/debian/rules
index 665233e..288b405 100755
--- a/beav-1.40/debian/rules
+++ b/beav-1.40/debian/rules
@@ -5,11 +5,18 @@
 CFLAGS = -O2 -g -Wall -DUNIX -DNOPROTO
 LDFLAGS = -s
 ARCH = $(shell dpkg --print-installation-architecture)
-STRIP=strip --strip-unneeded --remove-section=.note --remove-section=.comment
+ifneq ($(DEB_BUILD_GNU_TYPE),$(DEB_HOST_GNU_TYPE))
+	STRIP=$(DEB_HOST_GNU_TYPE)-strip
+	CC=$(DEB_HOST_GNU_TYPE)-gcc
+else
+	STRIP=strip
+	CC=gcc
+endif
+STRIP += --strip-unneeded --remove-section=.note --remove-section=.comment
 
 build:
 # Builds the binary package.
-	make CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)"
+	make CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" CC=$(CC)
 	touch build
 
 clean:
